store/catalog.go

```diff
@@ -178,17 +179,43 @@ func convertTable(table *api.Table) *catalog.Table {
 
 func converIndexList(list []*api.Index) []*catalog.Index {
 	var res []*catalog.Index
+	// sort index list by (name, position)
+	sort.Slice(list, func(i, j int) bool {
+		if list[i].Name != list[j].Name {
+			return list[i].Name < list[j].Name
+		}
+		return list[i].Position < list[j].Position
+	})
+
+	startPos := 0
```

Comment:
startPos and currentPos look like C-style and could be brittle. Do we have other algorithms such as mapping from index name to indices first and processing the indices with the same name?